frontends/web/src/routes/device/bitbox02/passphrase.tsx

```diff
@@ -184,147 +147,58 @@ export const Passphrase = ({ deviceID }: TProps) => {
-            {t('button.back')}
-          </Button>
-        </ViewButtons>
-      </View>
-    );
-  case 3:
-    return (
-      <View
-        key="step-why"
-        fullscreen
-        minHeight={CONTENT_MIN_HEIGHT}
-        onClose={onClose}
-        verticallyCentered>
-        <ViewHeader title={t('passphrase.why.title')} />
-        <ViewContent>
-          <MultilineMarkup tagName="p" markup={t('passphrase.why.message')} />
-        </ViewContent>
-        <ViewButtons>
-          <Button primary onClick={onContinue}>
-            {t('passphrase.considerations.button')}
-          </Button>
-          <Button secondary onClick={onBack}>
-            {t('button.back')}
-          </Button>
-        </ViewButtons>
-      </View>
-    );
-  case 2:
-    return (
-      <View
-        key="step-considerations"
-        fullscreen
-        minHeight={CONTENT_MIN_HEIGHT}
-        onClose={onClose}
-        verticallyCentered>
-        <ViewHeader title={t('passphrase.considerations.title')} />
-        <ViewContent>
-          <MultilineMarkup tagName="p" markup={t('passphrase.considerations.message')} />
-        </ViewContent>
-        <ViewButtons>
-          <Button primary onClick={onContinue}>
-            {t('passphrase.how.button')}
-          </Button>
-          <Button secondary onClick={onBack}>
-            {t('button.back')}
-          </Button>
-        </ViewButtons>
-      </View>
-    );
-  case 1:
-    return (
-      <View
-        key="step-how"
-        fullscreen
-        minHeight={CONTENT_MIN_HEIGHT}
-        onClose={onClose}
-        verticallyCentered>
-        <ViewHeader title={t('passphrase.how.title')} />
+  const handleBack = () => {
+    if (infoStep <= 0) {
+      handleAbort();
+      return null;
+    }
+    setInfoStep((infoStep) => infoStep - 1);
+  };
+
+  const handleContinue = () => {
+    if (infoStep === 5) {
+      setPassphrase(true);
+      return null;
+    }
+    setInfoStep((infoStep) => infoStep + 1);
+  };
+
+  type TStepData = { titleKey: string; messageKey: string; buttonTextKey: string; }[]
+
+  const stepData: TStepData = [
+    { titleKey: 'passphrase.intro.title', messageKey: 'passphrase.intro.message', buttonTextKey: 'passphrase.what.button' },
+    { titleKey: 'passphrase.what.title', messageKey: 'passphrase.what.message', buttonTextKey: 'passphrase.why.button' },
+    { titleKey: 'passphrase.why.title', messageKey: 'passphrase.why.message', buttonTextKey: 'passphrase.considerations.button' },
+    { titleKey: 'passphrase.considerations.title', messageKey: 'passphrase.considerations.message', buttonTextKey: 'passphrase.how.button' },
+    { titleKey: 'passphrase.how.title', messageKey: 'passphrase.how.message', buttonTextKey: 'passphrase.summary.button' },
+    { titleKey: 'passphrase.summary.title', messageKey: 'passphrase.summary.understandList', buttonTextKey: 'passphrase.enable' },
+  ];
+
+  const renderSteps = (idx: number) => {
+    return (<View
```

Comment:
We don't do it everywhere consistently but for readability 
```suggestion
    return (
      <View
```